lib/spack/spack/test/environment_modifications.py

```diff
@@ -62,16 +89,38 @@ def test_exclude_paths_from_inspection():
     assert len(env) == 0
 
 
+def make_pathlist(paths):
+    """Makes a fake list of platform specific paths"""
+    return os.pathsep.join(
+        [
+            driver + os.path.join(*path) if isinstance(path, list) else driver + path
```

Comment:
It's slightly confusing to refer to this as driver on windows, so I'd prefer  to avoid that, and also to consolidate all uses of `driver` outside of this into something like a `make_path` function (and `make_pathlist` can call that on each element of a list).

Reply:
Refactored.